mlir/lib/Dialect/Rock/Transforms/GemmToGridwise.cpp

```diff
@@ -130,11 +137,183 @@ computeGridSizeAttentionGemmElmtGemm(ConversionPatternRewriter &rw, Op op,
   return success();
 }
 
+static FailureOr<std::tuple<Value, Value, Value, Value>>
+arrangeGemmGemmSplitKTransform(OpBuilder &builder,
+                               RockGemmGemmWrapperInterface op, Location loc,
+                               const BufferDependencyAnalysis &bufferDeps,
+                               int64_t splitNFactor, Value a, Value b, Value c,
+                               Value out) {
+  // adjust the store method
+  auto storeMethod =
+      builder.getAttr<rock::StoreMethodAttr>(rock::StoreMethod::AtomicAdd);
+  op.setStoreMethodAttr(storeMethod);
+
+  // set the prefill attribute
+  auto func = llvm::cast<func::FuncOp>(op->getParentOp());
+  FailureOr<SmallVector<BlockArgument>> args =
+      traceGemmOutputToArgs(out, func, bufferDeps);
+  if (failed(args)) {
+    return op->emitError("can't trace gemm output to output argument");
+  }
+
+  auto attrName = rock::PrefillAttr::getMnemonic();
+  for (auto arg : args.value()) {
+    // initialize to zeros
+    auto elementType = cast<MemRefType>(arg.getType()).getElementType();
+    Attribute zero;
+    if (llvm::isa<FloatType>(elementType)) {
+      zero = builder.getFloatAttr(elementType, 0.0f);
+    } else if (llvm::isa<IntegerType>(elementType)) {
+      zero = builder.getIntegerAttr(elementType, 0);
+    } else {
+      return op->emitError("expecting `float` or `int` element type");
+    }
+    func.setArgAttrs(arg.getArgNumber(), builder.getNamedAttr(attrName, zero));
+  }
+
+  const int64_t origN = cast<MemRefType>(b.getType()).getShape()[2];
+  const int64_t nPad =
+      splitNFactor - math_util::mod_1_to_n(origN, splitNFactor);
+
+  b = padMatrix(b, builder, loc, "gemmK", 0, "gemmN", nPad);
```

Comment:
```suggestion
  b = padMatrix(b, builder, loc, "gemmK", 0, "gemmM", nPad);
```
Shuldn't this be M ?

Reply:
I think generally we define A -> MxK, B -> NxK. See rocmlir-gen (getGemmTypes). This is B, so it's N.